File: netcore/Flights/Flights.Core/Interfaces/IFlightRepository.cs
using Flights.Core.Model;
using System.Collections.Generic;

namespace Flights.Core.Interfaces
{
    public interface IFlightRepository
    {
        Flight Get(string id);
        List<Flight> GetAll(string query);
        void Insert(Flight flight);
        void Update(Flight flight);
        void Delete(string id);
    }
}

File: Java/src/main/java/flights/core/interfaces/IFlightRepository.java
package flights.core.interfaces;

import flights.core.model.Flight;
import java.util.List;

public interface IFlightRepository {
  Flight Get(String id);
  List<Flight> GetAll(String query);
  Flight Insert(Flight flight);
  Flight Update(Flight flight);
  void Delete(String id);
}
